internal/controller/eventbus/eventlog/eventlog.go

```diff
@@ -336,17 +335,20 @@ func (mgr *eventlogManager) UpdateSegment(ctx context.Context, m map[string][]Se
 	}
 }
 
-func (mgr *eventlogManager) GetEventLogSegmentList(elID vanus.ID) []*Segment {
-	result := make([]*Segment, 0)
+func (mgr *eventlogManager) GetEventLogSegmentList(elID vanus.ID) []Segment {
+	result := make([]Segment, 0)
 	v, exist := mgr.eventLogMap.Load(elID.Key())
 	if !exist {
 		return result
 	}
+
 	el, _ := v.(*eventlog)
-	s := el.head()
-	for s != nil {
-		result = append(result, s)
-		s = el.nextOf(s)
+	list := el.getAllSegments()
```

Comment:
why `getAllSegments` returns `[]*Segment` and make a `[]Segment` from it by copy with another locking?

Reply:
you can see L138, we have to save a point of Segment to map, and make a copy to make sure it's immutable for external access